File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> numbers;

        for(int i=0; i<nums.size(); i++){
            int diff = target-nums[i];
            if(numbers.find(diff)!=numbers.end()){
                auto it = numbers.find(diff);
                return {it->second, i};
            } else{
                numbers[nums[i]]=i;
            }
        }
        //error
        return {-1, -1};
    }
};

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {} #index, num

        for i, n in enumerate(nums):
            diff = target - n
            if diff in seen:
                return [seen[diff], i]
            else:
                seen[n] = i

